bolt-sidecar/src/signer/keystore.rs

```diff
@@ -57,6 +59,17 @@ impl KeystoreSigner {
         Ok(Self { keypairs, chain })
     }
 
+    /// Returns the public keys of the keypairs in the keystore.
+    pub fn pubkeys(&self) -> Vec<BlsPublicKey> {
+        self.keypairs
+            .iter()
+            .map(|kp| {
+                BlsPublicKey::try_from(kp.pk.serialize().to_vec().as_ref()).expect("valid pubkey")
+            })
+            .collect::<Vec<_>>()
+    }
```

Comment:
Related to the comment above, this should be an easy change
```suggestion
    /// Returns the public keys of the keypairs in the keystore.
    pub fn pubkeys(&self) -> HashSet<BlsPublicKey> {
        self.keypairs
            .iter()
            .map(|kp| {
                BlsPublicKey::try_from(kp.pk.serialize().to_vec().as_ref()).expect("valid pubkey")
            })
            .collect::<HashSet<_>>()
    }
```

Reply:
yep, done as suggested above

Reply:
1 quick point, for small collections a Vec is generally faster than a HashSet for iterating and finding apparently: https://gist.github.com/daboross/976978d8200caf86e02acb6805961195?permalink_comment_id=4400446#gistcomment-4400446